### DISCUSSION TITLE
how can i make the bb3 model like flask api or websocket api

### DISCUSSION THREAD

**Discussion by lonnieLuPacvue** [facebookresearch/ParlAI#5083] [Q&A] [answered]
i can run the bb3 model through below command
parlai safe_interactive --model-file zoo:bb3/bb3_3B/model --init-opt gen/r2c2_bb3

But i want make them like flask api or websocket,is there any solutions?
i read the terminal websocket and rewrite the config.yml like

tasks:
  default:
    onboard_world: MessengerBotChatOnboardWorld
    task_world: MessengerBotChatTaskWorld
    timeout: 1800
    agents_required: 1
task_name: chatbot
world_module: parlai.chat_service.tasks.chatbot.worlds
overworld: MessengerOverworld
max_workers: 30
opt:
  debug: True
  models:
    bb3_3B:
      model: transformer/generator
      model_file: zoo:bb3/bb3_3B/model
      interactive_mode: True
      # no_cuda: False
      override:
              #init_opt: arch/r2c2_base_3B
        search_server: 172.28.107.241:8080
        memory_decision: always
        search_decision: always
additional_args:
  page_id: 1

But i can chat with model.But the answer is all wrong.
Help~

**Comment by mojtaba-komeili** [comment, marked as answer]
Check out our [interactive_web](https://github.com/facebookresearch/ParlAI/blob/main/parlai/scripts/interactive_web.py) script.

**Comment by lonnieLuPacvue** [comment]
@mojtaba-komeili thanks, it's worked for me.But I find that when I start two client and send the request at the same time.The model deals it one by one.So,how I can do something to make it to be concurrent.

**Reply by mojtaba-komeili** [reply]
Our code is not optimized for performance in production much. This simple script is just for running small demos and debugging the model. For a simple solution, I recommend running multiple copies (maybe each inside a container). Or you could look into making it async.

**Reply by lonnieLuPacvue** [reply]
yes, I understand.But i want to now,Does this model can deal something with parallel

**Reply by mojtaba-komeili** [reply]
We do not provide that here. But we have knowledge of projects that were able to successfully do that.

**Reply by lonnieLuPacvue** [reply]
ok，thanks
